### PR TITLE
Sanitize Perm Checks

### PR DESCRIPTION
**Pull request by srkukarni**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

Tenant admin has the ability to produce/consume. This is honored by the http api, but not the binary api.
This pr brings consistency to this.

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by jiazhai** [comment]
@srkukarni  Is the superuser role has the same issue?

**Comment by srkukarni** [comment]
@jiazhai The superUser check happens in AuthorizationService so that is not affected.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java [hunk 1]
```diff
@@ -935,6 +937,38 @@ private SchemaData getSchema(PulsarApi.Schema protocolSchema) {
...
+    }
+
+    private CompletableFuture<Boolean> computeAuthorizedFuture(TopicName topicName, String role, String subscription, boolean produce) {
+        String path = PulsarWebResource.path(ConfigurationCacheService.POLICIES, topicName.getTenant());
+        try {
+            TenantInfo tenantInfo = service.getPulsar().getConfigurationCache().propertiesCache().get(path)
+                    .orElseThrow(() -> new PulsarServerException.NotFoundException("Tenant does not exist"));
+            CompletableFuture<Boolean> isTenantAdminFuture = service.getAuthorizationService().isTenantAdmin(topicName.getTenant(), role, tenantInfo, authenticationData);
+            return isTenantAdminFuture.thenCompose(isTenantAdmin -> {
+                if (isTenantAdmin) { return CompletableFuture.completedFuture(true); }
```
Comment (sijie) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java:955]: what is the motivation of this change? This seems to change the behavior of the current authorization system. It allows tenant admin to produce and consume messages. In general, this is a breaking change. There is also an attempt to change the authorization system in PIP-49. It was pushed back. We shouldn't change the existing authorization system. If you want to allow TenantAdmin to produce and consume messages, you should implement a new AuthorizationProvider.

**Comment by sijie** [comment]
> Tenant admin has the ability to produce/consume. This is honored by the http api, but not the binary api.

I am confused by this statement. Http doesn't provide any ability to produce and consume messages. You have to grant a role permissions to produce and consume messages using HTTP admin API.

**Comment by tisonkun** [comment]
Closed as stale and conflict. Please rebase and resubmit the patch if it's still relevant.

**Pull request closed without merging**